axelor-stock/src/main/java/com/axelor/apps/stock/db/repo/StockMoveManagementRepository.java

```diff
@@ -71,4 +85,51 @@ public void remove(StockMove entity) {
       super.remove(entity);
     }
   }
+
+  @Override
+  public Map<String, Object> populate(Map<String, Object> json, Map<String, Object> context) {
+    Long stockMoveId = (Long) json.get("id");
+    StockMove stockMove = find(stockMoveId);
+
+    if (stockMove.getStatusSelect() > STATUS_PLANNED || stockMove.getStockMoveLineList() == null) {
+      return super.populate(json, context);
+    }
+
+    int available = 0, missing = 0;
+    for (StockMoveLine stockMoveLine : stockMove.getStockMoveLineList()) {
+      if (stockMoveLine.getProduct() != null) {
+        StockLocationLine stockLocationLine = null;
+        if (stockMoveLine.getTrackingNumber() != null) {
+          stockLocationLine =
+              stockLocationLineService.getDetailLocationLine(
+                  stockMove.getFromStockLocation(),
+                  stockMoveLine.getProduct(),
+                  stockMoveLine.getTrackingNumber());
+        } else {
+          stockLocationLine =
+              stockLocationLineService.getStockLocationLine(
+                  stockMove.getFromStockLocation(), stockMoveLine.getProduct());
+        }
+        BigDecimal availableQty =
+            stockMoveLineService.getAvailableQty(stockMoveLine, stockLocationLine);
```

Comment:
The 'getAvailableQty' should only take 'stockMoveLine' and 'Location' as parameter and all above code of getting 'stockLocationLine' should be done that method, no needs do it here in populate method.